students/headieh_godwin/lesson05/mailroom.py

```diff
@@ -117,59 +94,54 @@ def valid_money():
         except ValueError:
             print("Invalid value")
 
+# Get name
+def valid_name():
+    while True:
+        r = input("Please enter a full name ")
+        r2= re.sub(r'[^a-zA-Z]+', '', r)
+        if r == '':
+            print('No name entered')
+        elif r2 == '':
+            print('Not a valid name')
+        else:
+            return r
+
+#quits
+def quits():
+    print("Bye!")
+    sys.exit()
+
 #Thanks!
 def thanks():
     """ Prompt for a donor name and amount - then prints email"""
-    response1 = input("Please enter a full name ")
-
-    #names2 = []
-    #for key in donors:
-    #    names2.append(donors[key]['name'])
-
-    #list comprehension
-    names = []
-    names = [donors[key]['name'] for key in donors]
-
+    response1 = valid_name()
+    names = [key for (key,value) in donors.items()]
```

Comment:
to get a list of names you can just do `list(donors)` and use it together with print > `print(list(donors))`